src/pudl_archiver/archivers/classes.py

```diff
@@ -271,6 +270,25 @@ async def get_hyperlinks(
             },
         )
         text = await retry_async(response.text)
+        return self.get_hyperlinks_from_text(text, filter_pattern)
+
+    def get_hyperlinks_from_text(
+        self,
+        text: str,
+        filter_pattern: typing.Pattern | None = None,
+    ) -> list[str]:
+        """Return all hyperlinks from HTML text.
+
+        This is a helper-helper function to perform very basic HTML-parsing functionality.
+        It extracts all hyperlinks from an HTML text, and returns those that match
+        a specified pattern. This means it can find all hyperlinks that look like
+        a download link to a single data resource.
+
+        Args:
+            text: text containing HTML.
+            filter_pattern: If present, only return links that contain pattern.
+        """
+        parser = _HyperlinkExtractor()
```

Comment:
I believe I have carefully sliced this to not conflict with [Marianne's `get_hyperlink` changes](https://github.com/catalyst-cooperative/pudl-archiver/pull/538/files) but I'll handle any massaging necessary if not